stdlib/jdbc/src/main/java/org/ballerinax/jdbc/statement/AbstractSQLStatement.java

```diff
@@ -174,6 +174,38 @@ TableValue constructTable(TableResourceManager rm, ResultSet rs, BStructureType
         return columnDefs;
     }
 
+    Object extractValueFromResultSet(ResultSetMetaData rsMeta, ResultSet rs, int index) throws SQLException {
+        Object value;
+        int columnType;
+        columnType = rsMeta.getColumnType(index);
+        switch (columnType) {
+        case Types.INTEGER:
+        case Types.TINYINT:
+        case Types.SMALLINT:
+        case Types.BIGINT:
+            value = rs.getLong(index);
+            break;
+        case Types.DOUBLE:
+            value = rs.getDouble(index);
+            break;
+        case Types.FLOAT:
+            value = rs.getFloat(index);
+            break;
+        case Types.BOOLEAN:
+        case Types.BIT:
+            value = rs.getBoolean(index);
+            break;
+        case Types.DECIMAL:
+        case Types.NUMERIC:
+            value = rs.getBigDecimal(index);
```

Comment:
Should we not create a `DecimalValue` out of the `BigDecimal` we are obtaining here?